src/world/GameWorld.cpp

```diff
@@ -37,7 +37,7 @@ GameWorld::GameWorld(const std::vector<PlayerInfo>& playerInfos, const GlobalGam
 {}
 
 /// Lädt eine Karte
-bool GameWorld::LoadMap(const std::string& mapFilePath, const std::string& luaFilePath)
+bool GameWorld::LoadMap(boost::weak_ptr<Game> game, const std::string& mapFilePath, const std::string& luaFilePath)
```

Comment:
Use a shared ptr here. Weak ptr semantic is 'can still contain a pointer but may not'. But here we actually want to ensure that we have a valid pointer.